### ISSUE TITLE
Error returned from introspection endpoint: An error occurred while sending the request.

### ISSUE THREAD

**Issue by unionthugface** [IdentityModel/IdentityModel.AspNetCore.OAuth2Introspection#185]
We have multiple applications using client credentials flow token introspection, communicating with a load-balanced solution using IdentityServer4; all of these are hosted in IIS.  At least once/day, one of the initiating apps will start generating the error `Error returned from introspection endpoint: An error occurred while sending the request` with callsite `IdentityModel.AspNetCore.OAuth2Introspection.OAuth2IntrospectionHandler.HandleAuthenticateAsync`.  No further exception is logged.  Sometimes just one app will start generating this error repeatedly, indicating that the problem is _not_ coming from the IdentityServer4 solution; other apps will have no trouble with token introspection at the same time.  Occasionally, all apps connecting to the IdentityServer4 solution will start producing this error, but that is more rare.  Failed request logs don't turn up anything, and neither does our APM.  Clients have told us they are unable to use our app(s) while this is happening.  It doesn't happen at the same time each day, and sometimes it happens multiple times per day.  We have had no luck figuring out what could be going on.  Any insight is greatly appreciated.

**Comment by steffibreuer** [comment]
We have the same problem.
@unionthugface have you already found a solution for this problem?

**Comment by unionthugface** [comment]
No, and we have never stopped trying to figure it out; it continues to elude us.  If you find something, please share, and I promise to do the same!

Sent from AOL on Android 
 
  On Thu, Sep 28, 2023 at 5:59 AM, ***@***.***> wrote:   


We have the same problem.
@unionthugface have you already found a solution for this problem?

**Comment by jakedansoncyber** [comment]
Seeing the same error on my end. This problem is hard to diagnose because this error only happens on 1% of requests. 

The problem seems to be originating from here, because I am having issues with multiple different IDPs, all with the same error.

**Comment by josephdecock** [comment]
Sorry to keep everyone waiting for a long time on this issue. We're in the process of migrating this repo into the Duende organization, and I'm trying to clean up all the outstanding issues. This one looks very tricky to reproduce, however we did recently merge a fix to some closely related code in #184. Once the migration to the Duende org is complete, we'll publish an updated build which hopefully will address these issues. If not, please open a discussion [here](https://github.com/orgs/DuendeSoftware/discussions).
